frontend/js/src/playlists/components/DuplicateTrackModal.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+
+type DuplicateTrackModalProps = {
+  message: string;
+  dontAskAgain: boolean;
+  setDontAskAgain: (value: boolean) => void;
```

Comment:
Instead of passing the setDontAskAgain down to the child component, let's instead make this modal return both booleans: the confirmation bool you already have, and also now the localDontAskAgain value.
Then in Playlist.tsx you can expect the modal to return two values (will leave a comment on the appropriate lines)